components/image_viewer/image_viewer.vue

```diff
@@ -0,0 +1,204 @@
+<template>
+  <div>
+    <dt-button
+      data-qa="dt-image-viewer-preview"
+      :tabindex="show ? -1 : 0"
+      importance="clear"
+      @click="open"
+    >
+      <img
+        class="d-wmn64 d-hmn64 w-wm332 d-hmx332"
+        :src="imageUrl"
+        :alt="imageAlt"
+      >
+    </dt-button>
+
+    <dt-lazy-show
+      transition="d-zoom"
+      :aria-hidden="isOpen"
+      :show="show"
+      class="d-modal"
+      data-qa="dt-modal"
+      v-on="modalListeners"
+      @mouseover="showCloseButton = true"
+      @mouseleave="showCloseButton = false"
+      @focusin=" showCloseButton = true"
+      @focusout=" showCloseButton = false"
+    >
+      <transition
+        appear
+        name="d-modal__dialog"
+      >
+        <div
+          v-show="show"
+          data-qa="dt-image-viewer-full"
+          :class="[
+            'd-modal__dialog', 'd-p0 d-bar0 d-wmx80p d-hmx80p d-image-viewer--modal',
+          ]"
+          role="dialog"
+          aria-modal="true"
+        >
+          <img
+            :src="imageUrl"
+            :alt="imageAlt"
+          >
+        </div>
+      </transition>
+      <dt-button
+        v-if="showCloseButton"
+        ref="closeImage"
+        data-qa="dt-image-viewer-close-btn"
+        class="d-modal__close"
+        circle
+        size="lg"
+        importance="clear"
+        kind="inverted"
+        :aria-label="closeAriaLabel"
+        @click="close"
+      >
+        <template #icon>
+          <dt-icon
+            name="close"
+            size="400"
+          />
+        </template>
+      </dt-button>
+    </dt-lazy-show>
+  </div>
+</template>
+
+<script>
+import {} from './image_viewer_constants.js';
+import Modal from '@/common/mixins/modal.js';
+import { EVENT_KEYNAMES } from '@/common/constants';
+import { DtIcon } from '@/components/icon';
+import { DtButton } from '@/components/button';
+import { DtLazyShow } from '@/components/lazy_show';
+
+export default {
+  name: 'DtImageViewer',
+
+  components: {
+    DtLazyShow,
+    DtButton,
+    DtIcon,
+  },
+
+  mixins: [Modal],
+
+  /* inheritAttrs: false is generally an option we want to set on library
+    components. This allows any attributes passed in that are not recognized
+    as props to be passed down to another element or component using v-bind:$attrs
+    more info: https://vuejs.org/v2/api/#inheritAttrs */
+  // inheritAttrs: false,
+
+  props: {
```

Comment:
Any way to set the size of the thumbnail? This seems like something that might need to be commonly done.

Reply:
Added imageButtonClass